### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by gaelL**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,43 +21,133 @@ This stack will deploy a php application on X Amazon EC2 instances behind an ALB
...
+|`cycloid_files_watched`|Provide log files you want to export to Cycloid logs.|`-`|`<Default log files watched>`|`False`|
+|`nginx_sites`|Contain Nginx vhosts to create on front servers. A default application and metrics vhosts are already provided.|`dict`|`<metric and application vhost>`|`False`|
+|`php_version_to_install`|PHP fpm version to install.|`-`|`7.2`|`False`|
 
-# Troubleshooting
-
-## Test ansible role with molecule
+# Molecule tests
 
 Requires a bucket which contains a build of magento sources and AWS accesskey
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
Requires a bucket which contains a build of magento sources and AWS access key
```

File: README.md [hunk 2]
```diff
@@ -1,17 +1,18 @@
...
-Linux operating system
-Nginx (Pronounced as Engine-X) web server
-MySQL (RDS) database server
-PHP-fpm for dynamic data processing
+This stack will create A full LEMP infrastructure based on ASG and RDS database.
 
+  * Linux operating system
+  * Nginx (Pronounced as Engine-X) web server
+  * MySQL (RDS) database server
+  * PHP-fpm for dynamic data processing
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
  * PHP-FPM for dynamic data processing
```

File: README.md [hunk 3]
```diff
@@ -1,17 +1,18 @@
...
+# stack-lemp (Linux, Nginx, MySQL, PHP)
 
-Linux operating system
-Nginx (Pronounced as Engine-X) web server
-MySQL (RDS) database server
-PHP-fpm for dynamic data processing
+This stack will create A full LEMP infrastructure based on ASG and RDS database.
 
+  * Linux operating system
+  * Nginx (Pronounced as Engine-X) web server
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
  * NGINX (Pronounced as Engine-X) web server
```

File: ansible/environments/front.yml.sample [hunk 4]
```diff
@@ -1 +1,10 @@
 # In this file you could override any variables present in default_front.yaml file
+
+#. nginx_sites (optional, dict): <metric and application vhost>
+#+ Contain Nginx vhosts to create on front servers. A default application and metrics vhosts are already provided.
+
+#. cycloid_files_watched (optional; dict): <Default log files watched>
+#+ Provide log files you want to export to Cycloid logs.
+
+#. php_version_to_install: 7.2
+#+ PHP fpm version to install.
```
Comment (sdurrheimer) [inline ansible/environments/front.yml.sample]:
```suggestion
#+ PHP-FPM version to install.
```

File: ansible/environments/front.yml.sample [hunk 5]
```diff
@@ -1 +1,10 @@
 # In this file you could override any variables present in default_front.yaml file
+
+#. nginx_sites (optional, dict): <metric and application vhost>
+#+ Contain Nginx vhosts to create on front servers. A default application and metrics vhosts are already provided.
```
Comment (sdurrheimer) [inline ansible/environments/front.yml.sample]:
```suggestion
#+ Contain NGINX vhosts to create on front servers. A default application and metrics vhosts are already provided.
```

File: pipeline/variables.sample.yml [hunk 6]
```diff
@@ -1,61 +1,114 @@
...
 config_git_private_key: ((git_config.ssh_key))
 
-lemp_git_repository: git@github.com:MyUser/code-magento.git
+
+#. lemp_git_repository (required): git@github.com:MyUser/code-lemp.git
+#+ Url to the git repository containing lemp website source code.
+lemp_git_repository: git@github.com:MyUser/code-lemp.git
+
+#. lemp_git_branch (required): master
+#+ Branch of the lemp source code git repository.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ Branch of the LEMP source code Git repository.
```

File: pipeline/variables.sample.yml [hunk 7]
```diff
@@ -1,61 +1,114 @@
...
 config_git_branch: master
+
+#. config_git_private_key (required): ((git_config.ssh_key))
+#+ SSH key pair to fetch the config git repository.
 config_git_private_key: ((git_config.ssh_key))
 
-lemp_git_repository: git@github.com:MyUser/code-magento.git
+
+#. lemp_git_repository (required): git@github.com:MyUser/code-lemp.git
+#+ Url to the git repository containing lemp website source code.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ URL to the Git repository containing LEMP website source code.
```

File: pipeline/variables.sample.yml [hunk 8]
```diff
@@ -1,61 +1,114 @@
...
+#. config_git_repository (required): git@github.com:MyUser/config-lemp-app.git
+#+ Git repository url containing the config of the stack.
 config_git_repository: git@github.com:MyUser/config-lemp-app.git
+
+#. config_git_branch (required): master
+#+ Branch of the config git repository.
 config_git_branch: master
+
+#. config_git_private_key (required): ((git_config.ssh_key))
+#+ SSH key pair to fetch the config git repository.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ SSH key pair to fetch the config Git repository.
```

File: pipeline/variables.sample.yml [hunk 9]
```diff
@@ -1,61 +1,114 @@
...
 # Repos
 #
 
-# Configuration of the git repository used to configure the stack
+#. config_git_repository (required): git@github.com:MyUser/config-lemp-app.git
+#+ Git repository url containing the config of the stack.
 config_git_repository: git@github.com:MyUser/config-lemp-app.git
+
+#. config_git_branch (required): master
+#+ Branch of the config git repository.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ Branch of the config Git repository.
```

File: pipeline/variables.sample.yml [hunk 10]
```diff
@@ -1,61 +1,114 @@
...
+#+ Password used by ansible vault to decrypt your vaulted files.
 ansible_vault_password: ((raw_ansible_vault_password))
 
-# Configuration of the bastion server used to run ansible (ssh_bastion depend of you credential name)
+#. bastion_url (required): admin@bastion.cycloid.io
+#+ bastion url used by ansible to connect on aws ec2 instances.
 bastion_url: admin@bastion.cycloid.io
+
+#. bastion_private_key_pair (required): ((ssh_bastion.ssh_key))
+#+ bastion SSH private key used by ansible to connect on aws ec2 instances and the bastion itself.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ bastion SSH private key used by ansible to connect on AWS EC2 instances and the bastion itself.
```

File: pipeline/variables.sample.yml [hunk 11]
```diff
@@ -1,61 +1,114 @@
...
 deploy_bucket_object_path: /catalog-lemp-app/($ environment $)/lemp-app.tar.gz
 
-# Password used for your ansible vault variables (raw_ansible_vault_password depend of you credential name).
+#. ansible_vault_password (required): ((raw_ansible_vault_password))
+#+ Password used by ansible vault to decrypt your vaulted files.
 ansible_vault_password: ((raw_ansible_vault_password))
 
-# Configuration of the bastion server used to run ansible (ssh_bastion depend of you credential name)
+#. bastion_url (required): admin@bastion.cycloid.io
+#+ bastion url used by ansible to connect on aws ec2 instances.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ bastion URL used by ansible to connect on AWS EC2 instances.
```

File: pipeline/variables.sample.yml [hunk 12]
```diff
@@ -1,61 +1,114 @@
...
+#. lemp_git_repository (required): git@github.com:MyUser/code-lemp.git
+#+ Url to the git repository containing lemp website source code.
+lemp_git_repository: git@github.com:MyUser/code-lemp.git
+
+#. lemp_git_branch (required): master
+#+ Branch of the lemp source code git repository.
 lemp_git_branch: master
+
+#. lemp_git_private_key (required): ((git_lemp_app.ssh_key))
+#+ SSH key pair to fetch lemp source code git repository.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ SSH key pair to fetch LEMP source code Git repository.
```

File: pipeline/variables.sample.yml [hunk 13]
```diff
@@ -1,61 +1,114 @@
...
+#+ Name of the project.
 project: ($ project $)
+
+#. customer (required): ($ organization_canonical $)
+#+ Name of the Cycloid Organization, used as customer variable name.
 customer: ($ organization_canonical $)
 
-# If you need to use a specific branch of stack
+#. stack_git_branch (required): master
+#+ Branch to use on the public stack git repository
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ Branch to use on the public stack Git repository
```

File: pipeline/variables.sample.yml [hunk 14]
```diff
@@ -1,61 +1,114 @@
...
 stack_git_branch: master
 
-# Path of your configuration
+#. config_ansible_path (required): ($ project $)/ansible
+#+ Path of Ansible files in the config git repository
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ Path of Ansible files in the config Git repository
```

File: terraform/lemp.tf.sample [hunk 15]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
+  #+ Amazon subnets IDs on which create each components.
   private_subnets_ids      = ["private-subnets"]
-  public_subnets_ids       = ["<public-subnets>"]
+
+  #. public_subnets_ids (required, array):
+  #+ Amazon subnets IDs on which create each components.
+  public_subnets_ids      = ["public-subnets"]
+
+  #. bastion_sg_allow (optional):
+  #+ Amazon source security group ID which will be allowed to connect on Fronts port 22 (ssh).
```
Comment (sdurrheimer) [inline terraform/lemp.tf.sample]:
```suggestion
  #+ Amazon source security group ID which will be allowed to connect on Fronts port 22 (SSH).
```

File: terraform/lemp.tf.sample [hunk 16]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
   bastion_sg_allow         = "<bastion-sg>"
 
-  # metrics_sg_allow         = "<prometheus-sg>"
+  #. metrics_sg_allow (optional): ""
+  #+ Additionnal security group ID to assign to Cycloid workers. Goal is to allow monitoring server to query metrics.
+  metrics_sg_allow        = "<prometheus-sg>"
+
+  #. keypair_name (requiredl): cycloid
+  #+ SSH keypair name to use to deploy ec2 instances.
```
Comment (sdurrheimer) [inline terraform/lemp.tf.sample]:
```suggestion
  #+ SSH keypair name to use to deploy EC2 instances.
```

File: terraform/lemp.tf.sample [hunk 17]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
-  # create_elasticache = true
-  # cache_subnet_group             = "<cache-subnet-id>" # If not specified, create a dedicated group with private_subnets_ids
-  # elasticache_type                 = "cache.t2.micro"
-  # elasticache_nodes                = "1"
-  # elasticache_parameter_group_name = "default.redis5.0"
-  # elasticache_engine_version = "5.0.0"
-  # elasticache_engine               = "redis"
-  # elasticache_port                 = "6379"
+  #. create_elasticache (optional, bool): true
+  # Define if we want to create or not an Amazon elasticache
```
Comment (sdurrheimer) [inline terraform/lemp.tf.sample]:
```suggestion
  # Define if we want to create or not an Amazon ElastiCache
```

File: terraform/lemp.tf.sample [hunk 18]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
 
-  ###
-  # S3 medias
-  ###
+  #. cache_subnet_group (optional):
+  #+ Name os the Amazon elasticache subnet group to use. If not specified, create a dedicated group with private_subnets_ids.
```
Comment (sdurrheimer) [inline terraform/lemp.tf.sample]:
```suggestion
  #+ Name os the Amazon ElastiCache subnet group to use. If not specified, create a dedicated group with private_subnets_ids.
```

File: terraform/lemp.tf.sample [hunk 19]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
+
+  #. elasticache_type (optional): cache.t2.micro
+  #+ Instance size to use for elasticache nodes.
```
Comment (sdurrheimer) [inline terraform/lemp.tf.sample]:
```suggestion
  #+ Instance size to use for ElastiCache nodes.
```

File: terraform/lemp.tf.sample [hunk 20]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
 
-  # create_s3_medias = true
+  #. elasticache_nodes (optional): 1
+  #+ Number of nodes in the elasticache cluster.
```
Comment (sdurrheimer) [inline terraform/lemp.tf.sample]:
```suggestion
  #+ Number of nodes in the ElastiCache cluster.
```

File: terraform/lemp.tf.sample [hunk 21]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
+
+  #. elasticache_parameter_group_name (optional): default.redis5.0
+  #+ Elasticache group parameter name to use.
```
Comment (sdurrheimer) [inline terraform/lemp.tf.sample]:
```suggestion
  #+ ElastiCache group parameter name to use.
```

File: terraform/lemp.tf.sample [hunk 22]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
+
+  #. elasticache_engine_version (optional): 5.0.0
+  #+ Version of the Elasticache engine
```
Comment (sdurrheimer) [inline terraform/lemp.tf.sample]:
```suggestion
  #+ Version of the ElastiCache engine
```

File: terraform/lemp.tf.sample [hunk 23]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
+
+  #. elasticache_engine (optional): redis
+  #+ Type of the elasticache engine
```
Comment (sdurrheimer) [inline terraform/lemp.tf.sample]:
```suggestion
  #+ Type of the ElastiCache engine
```

File: terraform/lemp.tf.sample [hunk 24]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
+
+  #. elasticache_port (optional): 6379
+  #+ Port of the elasticache
```
Comment (sdurrheimer) [inline terraform/lemp.tf.sample]:
```suggestion
  #+ Port of the ElastiCache
```

File: terraform/outputs.tf [hunk 25]
```diff
@@ -1,39 +1,48 @@
 output "alb_front_dns_name" {
-  value = "${module.lemp.alb_front_dns_name}"
+  value       = "${module.lemp.alb_front_dns_name}"
+  description = "DNS name of the front alb."
```
Comment (sdurrheimer) [inline terraform/outputs.tf]:
```suggestion
  description = "DNS name of the front ALB."
```

File: terraform/outputs.tf [hunk 26]
```diff
@@ -1,39 +1,48 @@
...
 }
 
 output "alb_front_zone_id" {
-  value = "${module.lemp.alb_front_zone_id}"
+  value       = "${module.lemp.alb_front_zone_id}"
+  description = "Zone ID of the front alb."
```
Comment (sdurrheimer) [inline terraform/outputs.tf]:
```suggestion
  description = "Zone ID of the front ALB."
```

File: terraform/outputs.tf [hunk 27]
```diff
@@ -1,39 +1,48 @@
...
 }
 
 output "elasticache_address" {
-  value = "${module.lemp.elasticache_address}"
+  value       = "${module.lemp.elasticache_address}"
+  description = "Address of the elasticache."
```
Comment (sdurrheimer) [inline terraform/outputs.tf]:
```suggestion
  description = "Address of the ElastiCache."
```

File: pipeline/variables.sample.yml [hunk 28]
```diff
@@ -1,61 +1,114 @@
...
 bastion_private_key_pair: ((ssh_bastion.ssh_key))
 
+
 #
 # Repos
 #
 
-# Configuration of the git repository used to configure the stack
+#. config_git_repository (required): git@github.com:MyUser/config-lemp-app.git
+#+ Git repository url containing the config of the stack.
```
Comment (sdurrheimer) [inline pipeline/variables.sample.yml]:
```suggestion
#+ Git repository URL containing the config of the stack.
```

File: README.md [hunk 29]
```diff
@@ -20,43 +21,133 @@ This stack will deploy a php application on X Amazon EC2 instances behind an ALB
   * **Elasticache** (optional): Amazon Elasticache (redis)
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
  * **ElastiCache** (optional): Amazon ElastiCache (Redis)
```

File: README.md [hunk 30]
```diff
@@ -1,17 +1,18 @@
...
 # Architecture
 
-This stack will deploy a php application on X Amazon EC2 instances behind an ALB load balancer, using RDS database and optional ElasticCache/s3 medias bucket.
+<p align="center">
+<img src="docs/diagram.png" width="400">
+</p>
 
-<img src="https://raw.githubusercontent.com/cycloid-community-catalog/stack-lemp/master/diagram.png" width="400">
 
   * **ALB**: Amazon application loadbalancer
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
  * **ALB**: Amazon Application Load Balancer
```

File: README.md [hunk 31]
```diff
@@ -1,17 +1,18 @@
...
   * **ASG**: Autoscaling group for fronts
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
  * **ASG**: Amazon Auto Scaling group for fronts
```

File: README.md [hunk 32]
```diff
@@ -1,17 +1,18 @@
-# stack-lemp
-(Linux, Nginx, MySQL, PHP)
+# stack-lemp (Linux, Nginx, MySQL, PHP)
```
Comment (sdurrheimer) [inline README.md]:
```suggestion
# stack-lemp (Linux, NGINX, MySQL, PHP)
```

File: README.md [hunk 33]
```diff
@@ -20,43 +21,133 @@ This stack will deploy a php application on X Amazon EC2 instances behind an ALB
...
+> **Note** The pipeline contains a manual approval between terraform plan and terraform apply.
+> That means if you trigger a terraform plan, to apply it, you have to go on terraform apply job
+> and click on the `+` button to trigger it.
 
-**Ansible**
+<img src="docs/pipeline.png" width="800">
+
+**Jobs description**
+
+  * `build-worker-ami` : Build Front Amazon image AMI using Packer and Ansible.
```
Comment (xlr-8) [inline README.md]: build-ami-front* it's the name of the job, I guess it's a wrong copy/paste

File: README.md [hunk 34]
```diff
@@ -20,43 +21,133 @@ This stack will deploy a php application on X Amazon EC2 instances behind an ALB
...
+  * `build-application`: Runs the appropriate php/composer commands and build a release file of the code to put it on S3.
+  * `unittest`: Dummy job meant to eventually be replaced by proper tests or removed.
```
Comment (xlr-8) [inline README.md]: unittests* according to the pipeline screenshot

File: README.md [hunk 35]
```diff
@@ -20,43 +21,133 @@ This stack will deploy a php application on X Amazon EC2 instances behind an ALB
...
+  * `terraform-plan`: Terraform job that will simply make a plan of the stack.
```
Comment (xlr-8) [inline README.md:48]: I'd put the `terraform-plan` before the `unittests` one as it is at the same level as the `buildapplication` pipeline wise, WDYT?

File: README.md [hunk 36]
```diff
@@ -20,43 +21,133 @@ This stack will deploy a php application on X Amazon EC2 instances behind an ALB
...
+  * `terraform-apply`: Terraform job similar to the plan one, but will actually create/update everything that needs to. Please see the plan diff for a better understanding.
+  * `application-deployment`: Simply trigger a deployment using Ansible of the last version of the code on existing instances.
+  * `functional-tests`: Dummy job meant to eventually be replaced by proper functional or removed.
+  * `terraform-destroy`: :warning: Terraform job meant to destroy the whole stack - **NO CONFIRMATION ASKED**. If triggered, the full project **WILL** be destroyed. Use with caution.
```
Comment (xlr-8) [inline README.md:53]: Shall we specify that it's not displayed in the pipeline screenshot, but is on a dedicated pipeline tab?

File: README.md [hunk 37]
```diff
@@ -20,43 +21,133 @@ This stack will deploy a php application on X Amazon EC2 instances behind an ALB
...
+
+## Ansible
 
   * Playbook and packer config to build a debian image with telegraf, fluentd, nginx and php-fpm installed
 
+|Name|Description|Type|Default|Required|
+|---|---|:---:|:---:|:---:|
+|`cycloid_files_watched`|Provide log files you want to export to Cycloid logs.|`-`|`<Default log files watched>`|`False`|
+|`nginx_sites`|Contain Nginx vhosts to create on front servers. A default application and metrics vhosts are already provided.|`dict`|`<metric and application vhost>`|`False`|
+|`php_version_to_install`|PHP fpm version to install.|`-`|`7.2`|`False`|
```
Comment (xlr-8) [inline README.md:148]: I assume all the tables of params for pipeline, terraform/ansible/ouptut was generated by your script? Or shall I check it too?

**Review by xlr-8 — CHANGES_REQUESTED** [review]
Nice format! Couple comments/questions

File: pipeline/variables.sample.yml [hunk 38]
```diff
@@ -1,61 +1,114 @@
+  
 #
 # Terraform
 #
 
 # Amazon access to use to run terraform. Usually admin account to be able to create Amazon services.
+
+#. aws_access_key (required): ((aws.access_key))
+#+ Amazon AWS access key for Terraform. see value format [Here](https://docs.cycloid.io/advanced-guide/integrate-and-use-cycloid-credentials-manager.html#vault-in-the-pipeline)
```
Comment (xlr-8) [inline pipeline/variables.sample.yml]: See* here*

File: pipeline/variables.sample.yml [hunk 39]
```diff
@@ -1,61 +1,114 @@
...
 aws_access_key: ((aws.access_key))
+
+#. aws_secret_key (required): ((aws.secret_key))
+#+ Amazon AWS secret key for Terraform. see value format [Here](https://docs.cycloid.io/advanced-guide/integrate-and-use-cycloid-credentials-manager.html#vault-in-the-pipeline)
```
Comment (xlr-8) [inline pipeline/variables.sample.yml]: See* here*

File: pipeline/variables.sample.yml [hunk 40]
```diff
@@ -1,61 +1,114 @@
...
 
-# Password used for your rds. Set "empty" if you dont use databases
-rds_password: ((raw_rds_password))
+#. aws_default_region (required): eu-west-1
+#+ Amazon AWS region to use for Terraform.
+aws_default_region: eu-west-1
 
-# Bucket configuration to store terraform remote state file.
+#. terraform_storage_bucket_name (required): ($ organization_canonical $)-terraform-remote-state
+#+ AWS S3 bucket name to store terraform remote state file.
```
Comment (xlr-8) [inline pipeline/variables.sample.yml:21]: Shall we re-comment that those variables `($ organization_canonical $)` are replaced by the product, or shall we simply point to our documentation link?

File: README.md (hunk 37 above)
Comment (gaelL) [reply README.md:148]: yes tabs are generated by my scripts. From comments in the code

File: pipeline/variables.sample.yml [hunk 41]
```diff
@@ -1,61 +1,114 @@
...
 deploy_bucket_object_path: /catalog-lemp-app/($ environment $)/lemp-app.tar.gz
 
-# Password used for your ansible vault variables (raw_ansible_vault_password depend of you credential name).
+#. ansible_vault_password (required): ((raw_ansible_vault_password))
+#+ Password used by ansible vault to decrypt your vaulted files.
 ansible_vault_password: ((raw_ansible_vault_password))
 
-# Configuration of the bastion server used to run ansible (ssh_bastion depend of you credential name)
+#. bastion_url (required): admin@bastion.cycloid.io
+#+ bastion URL used by ansible to connect on AWS EC2 instances.
```
Comment (xlr-8) [inline pipeline/variables.sample.yml:50]: Shall we set this to such default value? As it's required might not make sense?

File: terraform/lemp.tf.sample [hunk 42]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
+  #+ Amazon subnets IDs on which create each components.
+  public_subnets_ids      = ["public-subnets"]
+
+  #. bastion_sg_allow (optional):
+  #+ Amazon source security group ID which will be allowed to connect on Fronts port 22 (SSH).
   bastion_sg_allow         = "<bastion-sg>"
 
-  # metrics_sg_allow         = "<prometheus-sg>"
+  #. metrics_sg_allow (optional): ""
+  #+ Additionnal security group ID to assign to Cycloid workers. Goal is to allow monitoring server to query metrics.
```
Comment (xlr-8) [inline terraform/lemp.tf.sample]: This doesn't seem to be for the `workers`, might want to update it, and regenerate the comments

File: terraform/lemp.tf.sample [hunk 43]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
+  metrics_sg_allow        = "<prometheus-sg>"
+
+  #. keypair_name (requiredl): cycloid
+  #+ SSH keypair name to use to deploy EC2 instances.
```
Comment (xlr-8) [inline terraform/lemp.tf.sample]: Shall we precise that this ssh key, has to exist on the AWS console?

File: terraform/lemp.tf.sample [hunk 44]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
-  # rds_storage_type     = "gp2"
-  # rds_subnet_group     = "<rds-subnet-id>" # If not specified, create a dedicated group with private_subnets_ids
-  # rds_type             = "db.t3.small"
-  # rds_username         = "magento"
-  # rds_engine           = "mysql"
-  # rds_engine_version   = "5.7.16"
-  # rds_backup_retention = 7
-  # rds_parameters       = "default.mysql5.7"
+  #. create_rds (optional, bool): false
+  #+ Define if we want to create or not a RDS database.
```
Comment (xlr-8) [inline terraform/lemp.tf.sample]: an RDS

File: terraform/lemp.tf.sample [hunk 45]
```diff
@@ -7,65 +7,130 @@ module "lemp" {
...
-  # elasticache_engine               = "redis"
-  # elasticache_port                 = "6379"
+  #. create_elasticache (optional, bool): true
+  # Define if we want to create or not an Amazon ElastiCache
 
-  ###
-  # S3 medias
-  ###
+  #. cache_subnet_group (optional):
+  #+ Name os the Amazon ElastiCache subnet group to use. If not specified, create a dedicated group with private_subnets_ids.
```
Comment (xlr-8) [inline terraform/lemp.tf.sample]: of*

**Review by xlr-8 — COMMENTED** [review]
Sorry didn't refresh everything :(

File: README.md (hunk 36 above)
Comment (gaelL) [reply README.md:53]: I think that's ok if this job is little bit harder to found :p

File: pipeline/variables.sample.yml (hunk 41 above)
Comment (gaelL) [reply pipeline/variables.sample.yml:50]: ok let's put something better as example

File: pipeline/variables.sample.yml (hunk 40 above)
Comment (gaelL) [reply pipeline/variables.sample.yml:21]: If it's not clear, it will be up to us to make it more clear in the doc or frontend display

**Pull request merged**